API_GUIDE.md

```diff
@@ -176,3 +174,101 @@ In this report, any counter that starts with `aten::`
 indicates a context switch between the XLA device and CPU, which can be a
 potential performance optimization area in the model code.
 
+# Debugging
```

Comment:
I thought we agreed on having a `TROUBLESHOOTING` document with both debugging and perf tuning suggestions ...

Reply:
Yea I plan to do that after this. 
Or I can do it directly on this PR ;) Thought this will be easier to review since in `TROUBLESHOOTING` I might want to add/modify things as well :P

Reply:
Why 2 PR?
Let's do it here, and if we have to add more later, we do it in a new PR.